pkg/strings/all_test.go

```diff
@@ -1,86 +1,84 @@
 package strings
 
 import (
-	"fmt"
 	"testing"
 )
 
 func TestMatchAndCaptures(t *testing.T) {
 	var tests = []struct {
 		pattern string
 		subject string
-		matches bool
-		invalid bool
+		matches []string
 	}{
 		{
 			pattern: "|.*|",
 			subject: "one |two| three |four| five",
-			// [|two| three |four|]
+			matches: []string{"|two| three |four|"},
 		},
 		{
 			pattern: "|.+|",
 			subject: "one |two| three |four| five",
-			// [|two| three |four|]
+			matches: []string{"|two| three |four|"},
 		},
 		{
 			pattern: "|.-|",
 			subject: "one |two| three |four| five",
-			// [|two|]
+			matches: []string{"|two|"},
 		},
 		{
 			pattern: "%$(.-)%$",
 			subject: "4+5 = $return 4+5$",
-			// [return 4 + 5]
+			// matches: []string{"return 4 + 5"},
+			matches: []string{"$return 4+5$", "return 4+5"},
```

Comment:
@fibonacci1729 a few of these need a closer check

Reply:
@adamreese thank you! huge help!